content/docs/user-guide/visualization/environments/terrain/terrain-developer-guide/architecture.md

```diff
@@ -0,0 +1,143 @@
+
+## Single system, multiple components
+
+Another key aspect to the terrain system design is that it provides a singleton system as a single entry point for all terrain queries and management of global terrain settings. However, all of the terrain data and components that feed the system are split apart into discrete pieces. There are many reasons for both of these design decisions.
+
+Single terrain system:
+
+* Defines a single "world grid" for the entire terrain, which presents a consistent resolution and alignment for the terrain grid everywhere. This makes it easy to query the system, since it's possible to walk through the world at consistently-spaced positions everywhere, regardless of the input data resolution and positioning.
+  * This also means the renderer doesn't need to solve seams with adjacent meshes of different resolutions, manage LODs across different resolutions, deal with overlapping terrain meshes, etc.
+* Provides a way to manage priorities for overlapping terrain data regions.
+* Gives a single well-defined API connection point for queries and notifications, without forcing everything using terrain to know about all of the individual data pieces that have been used to assemble the terrain.
+
+Multiple terrain components:
+
+* The world can be divided into arbitrary rectangular regions via **Terrain Layer Spawners**. The rectangles can be different sizes, and areas of the world without terrain can remain empty.
+  * This enables multiple content creators to work on different regions at the same time, since they're working on physically separated data.
+  * Data usage can be optimized by only placing data where it's needed.
+  * Regions can be dynamically loaded and unloaded individually, making it easy to implement coarse-grained terrain streaming.
+* Each region can have different subsets of features.
+  * Regions where gameplay occurs can spawn terrain physics representations, but distant terrain regions that are for visuals only don't need to have physics enabled.
+  * Aspects of terrain rendering can be disabled for server-side or machine learning simulations that don't require a visual representation of terrain.
+* Different components can have different input data resolutions. The terrain system defines a single _output_ terrain grid and resolution, but the input data resolutions can be varied to optimize the data size and complexity based on the need for that region. Even within a region, two different data providers (such as height and surface type) can use two different resolutions based on what's appropriate for that type of data.
+* The functionality can be replaced at a component-by-component level.
+  * The PhysX Heightfield component could be replaced with one that connects to another physics system such as Jolt or Bullet.
```

Comment:
Those links are correct, but I don't see any obvious trademark information. Would it be correct to refer to them as "Jolt Physics" and "Bullet Physics SDK", since that's how they are referred to on their main github pages?

Reply:
Yes! Using the name they use to refer to themselves is a safe bet.